test/test_main.py

```
@@ -186,6 +190,51 @@ def test_fetch_latest_release_writes_cache(
+
+    payload = b'{"html_url": "https://example.invalid", "tag_name": "v2.0.0"}'
+    urlopen_mock = mocker.patch(
+        "ansiblelint.config.urllib.request.urlopen",
+        return_value=BytesIO(payload),
+    )
+    config._fetch_latest_release(str(tmp_path / "latest.json"))  # ruff:ignore[private-member-access]
+    assert urlopen_mock.call_args.kwargs.get("timeout") == 10
+
+    # A timeout while establishing the connection.
+    urlopen_mock.side_effect = TimeoutError
+    assert config._fetch_latest_release(str(tmp_path / "fail.json")) == {}  # ruff:ignore[private-member-access]
```

Ruff (0.16.0)

[warning] Invalid rule code in suppression: private-member-access

Enable `lint.preview` to use rule names

(RUF102)